fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/TemplateDruidQuery.java

```diff
@@ -43,7 +44,7 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(TemplateDruidQuery.class);
 
-    private final TemplateDruidQuery nestedQuery;
+    private final Optional<TemplateDruidQuery> nestedQuery;
```

Comment:
It's generally considered best practice not to wrap a field in `Optional`. Instead, the field is the raw value, and the guts of the class deal with `null`. Then, when you want to return a reference to the value, you wrap the return value in an `Optional`: https://zeroturnaround.com/rebellabs/java-8-best-practices-cheat-sheet/